extension/src/XLIFFDocument.ts

```diff
@@ -196,6 +196,50 @@ export class Xliff implements XliffDocumentInterface {
     public sortTransUnits() {
         this.transunit.sort(CompareTransUnitId);
     }
+
+    /**
+     * Returns an array of trans-units where source matches and the translation differs from the input TransUnit
+     * @param transUnit trans-unit to match with.
+     * @returns TransUnit[]
+     */
+    public getSameSourceDifferentTarget(transUnit: TransUnit): TransUnit[] {
+        return this.transunit.filter(t => ((t.source === transUnit.source) && (t.targetTextContent !== transUnit.targetTextContent)));
+    }
+
+    /**
+     * Checks if source exists more than once.
+     * @param source Source string to search for.
+     * @returns boolean
+     */
+    public sourceHasDuplicates(source: string): boolean {
+        return this.getTransUnitsBySource(source).length > 1;
+    }
+
+    /**
+     * Returns an array of trans-units where source matches the input.
+     * @param source Source string to search for.
+     * @returns TransUnit[]
+     */
+    public getTransUnitsBySource(source: string): TransUnit[] {
+        return this.transunit.filter(t => t.source === source);
+    }
+
+    /**
+     * Returns an array of trans-units with matching sources and different translations.
+     * @returns TransUnit[]
+     */
+    public differentlyTranslatedTransunits(): TransUnit[] {
+        let transUnits: TransUnit[] = [];
+        this.transunit.forEach(tu => {
+            this.getSameSourceDifferentTarget(tu).forEach(duplicate => {
+                if (transUnits.filter(a => a.id === duplicate.id).length === 0) {
+                    transUnits.push(duplicate);
+                }
+            });
+        });
+        return transUnits;
```

Comment:
@jwikman I couldn't help thinking there would be a one liner for this and of course I found a way. It's neat, but I kind of feel that it's harder to read and it doesn't really feel like it belongs. What's your spontaneous reaction when you read it? 
```suggestion
        return [...new Set(this.transunit.map(t => this.getSameSourceDifferentTarget(t)).flat())];
```
Set returns unique array and flat() flattens what else would've been a 2d array.

Reply:
It's cool that it is even possible.
But I prefer the first one, as you might have expected ;)